cmd/esmapping-generator/app/renderer/render.go

```diff
@@ -34,11 +35,37 @@ func GetMappingAsString(builder es.TemplateBuilder, opt *app.Options) (string, e
 		UseILM:          enableILM,
 		ILMPolicyName:   opt.ILMPolicyName,
 	}
-	return mappingBuilder.GetMapping(opt.Mapping)
+
+	mappingType, err := stringToMappingType(opt.Mapping)
+	if err != nil {
+		return "", err
+	}
+
+	return mappingBuilder.GetMapping(mappingType)
 }
 
 // IsValidOption checks if passed option is a valid index template.
 func IsValidOption(val string) bool {
```

Comment:
I don't think this function is needed. Wherever it's called the called can instead do
```
if m, err := MappingTypeFromString(val); err == nil {
  // 
}
```

Reply:
Not addressed